src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/BotSegmentsPlaybackController.cs

```diff
@@ -192,7 +209,7 @@ private void ProcessBotSegments()
                         {
                             if (_nextBotSegments.Count < 2)
                             {
-                                var next = _dataPlaybackContainer.DequeueBotSegment();
+                                var next = _dataPlaybackContainer.DequeueBotSegment(out _botSegmentListValidations);
```

Comment:
@RG-nAmKcAz quick question on this actually regarding the lookahead logic. Does this mean that the currently running validation will be lost and replaced? I am not full understanding this code yet but I wanted to check if this means that validations that are currently running and not yet finished may then be running forever, since now we only operate on these new validations. Maybe we don't need to worry about that because we run StopValidations on the data container itself which does hold onto the reference...

Reply:
i'll huddle with you on this

Stop I think we're fine

but you correctly point out that we may not be considering all the proper validations... I want to discuss if we get rid of this look ahead logic all together or not